test/e2e/windows/service.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package windows
+
+import (
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/pkg/securitycontext"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	defaultCmd          = "curl.exe -s -o 127.0.0.1"
+	defaultWindowsImage = "mcr.microsoft.com/windows/nanoserver:1809"
+)
+
+var (
+	defaultWindowsPod = &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "windows-tester-",
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:            "ping-pod",
+					Image:           defaultWindowsImage,
+					Command:         []string{"cmd"},
+					Args:            []string{"-c", defaultCmd},
+					SecurityContext: securitycontext.ValidSecurityContextWithContainerDefaults(),
+				},
+			},
+			RestartPolicy: v1.RestartPolicyOnFailure,
+		},
+	}
+)
+
+var _ = SIGDescribe("Services", func() {
+	f := framework.NewDefaultFramework("services")
+
+	var cs clientset.Interface
+
+	BeforeEach(func() {
+		cs = f.ClientSet
+	})
+	//Note(immuzz -> Muzz Imam ) This test will run only for Windows Containers
+	It("should be able to create a functioning NodePort service for Windows", func() {
+		serviceName := "nodeport-test"
+		ns := f.Namespace.Name
+
+		jig := framework.NewServiceTestJig(cs, serviceName)
+		nodeIP := framework.PickNodeIP(jig.Client) // for later
+
+		By("creating service " + serviceName + " with type=NodePort in namespace " + ns)
+		service := jig.CreateTCPServiceOrFail(ns, func(svc *v1.Service) {
+			svc.Spec.Type = v1.ServiceTypeNodePort
+		})
+		jig.SanityCheckService(service, v1.ServiceTypeNodePort)
+		nodePort := int(service.Spec.Ports[0].NodePort)
+
+		By("creating pod to be part of service " + serviceName)
+		jig.RunOrFail(ns, nil)
+		
+		// Create a testing pod so we can ping the nodeport from within the cluster
+		By(fmt.Sprintf("creating testing pod to curl http://%s:%d", nodeIP, nodePort))
+		windowsSelector := map[string]string{"beta.kubernetes.io/os": "windows"}
+		winPodSpec := defaultWindowsPod.DeepCopy()
+		winPodSpec.Spec.NodeSelector = windowsSelector
+		curl := fmt.Sprintf("curl.exe -s -o /dev/null -w \"%%{http_code}\" http://%s:%d", nodeIP, nodePort)
```

Comment:
there is no `/dev/null` on Windows, but there is `NUL`

i'm going to assume we are trying to get the HTTP status?
try:
```
curl := fmt.Sprintf(`curl.exe -s -o NUL -w "%{http_code}" http://%s:%d`, nodeIP, nodePort)
```

but one question - if we are duplicating the Pod and always overwriting the default command, why do we need the default command?

Reply:
At the beginning we were using the default command just to put a sleep and be able to exist. You are right, if this is not used in other context there is no need to set it up. Removing it.

Reply:
Also:
 
```
-o /dev/null
```

And 

```
-o NUL
```

both behave the same way on both `powershell` and in `cmd`. But if you want we can adopt your way.

Reply:
doing `curl.exe -s -o /dev/null` will create the actual path e.g. `C:\dev\null`.

sadly, powershell and cmd use different variables for the null device:
PS uses `$null`
cmd uses `NUL`.

and one does not work for the other.
if this test is supposed to work for both shells, just pipe to a local file (but not /dev/null):

`curl.exe s -o ./curl-output`

/hold

Reply:
@neolit123  i am not seeing the same behavior. on my windows 10 laptop. when i run `curl -s -o /dev/null -w "%{http_code}" "https://img-s-msn-com.akamaized.net/tenant/amp/entityid/BBUmCRP.img?h=1224&w=799&m=6&q=60&o=f&l=f&x=1064&y=935"`, even with admin privileges the c:/dev/null folder is not created. the output is on the console

Reply:
Sorry guys the changes are done, but seems like @immuzz didnt give me contributor access to his repo. Also remember this is 1809 windows, which does behave differently.
Waiting for the access you can review the code here:
https://github.com/kubernetes/kubernetes/compare/master...immuzz:origin/nodeportwindows

Reply:
it's created for me on both Windows 7 and Windows 10.
but there is definitely no `/dev/null` on Windows:
https://stackoverflow.com/questions/313111/is-there-a-dev-null-on-windows

to be on the safe side and given we are using `cmd -c` we should just do what i recommended in the first comment:
```
curl.exe -s -o NUL -w "%{http_code}" http://%s:%d`, nodeIP, nodePort
```

Reply:
@brusMX if you'd like just close this PR using `/close` and open a new one with the changes.
but please squash the commits into a single one.

Reply:
> Also remember this is 1809 windows, which does behave differently.

i have 1809 here too:
![null-device-windows](https://user-images.githubusercontent.com/331852/53773978-44947380-3ef5-11e9-884e-004ee4f816e2.png)